src/warp.jl

```diff
@@ -0,0 +1,38 @@
+# None of these types are provided by this package, so the following line seems unclean
+#     @inline Base.getindex(A::AbstractExtrapolation, v::StaticVector) = A[convert(Tuple, v)...]
+# furthermore it would be ambiguous with getindex(::Extrapolation, xs...) after https://github.com/tlycken/Interpolations.jl/pull/141
+@inline _getindex(A, v::StaticVector) = A[convert(Tuple, v)...]
+
+warp(img::AbstractArray, args...) = warp(interpolate(img, BSpline(Linear()), OnGrid()), args...)
+
+function warp{T<:Colorant,S}(::Type{T}, img::AbstractArray{S}, args...)
+    TCol = typeof(convert(T,one(S)))
+    TNorm = eltype(TCol)
+    apad, pad = Interpolations.prefilter(TNorm, TCol, img, typeof(BSpline(Linear())), typeof(OnGrid()))
+    itp = Interpolations.BSplineInterpolation(TNorm, apad, BSpline(Linear()), OnGrid(), pad)
```

Comment:
What are the advantages of doing this here, rather than picking types for the storage output? That is, suppose it transiently computes an `RGB{Float64}`, that can still be stashed in an `RGB{N0f8}`.